kubernetes/data_source_kubernetes_config_map_test.go

```diff
@@ -0,0 +1,49 @@
+package kubernetes
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccKubernetesDataSourceConfigMap_basic(t *testing.T) {
+	name := fmt.Sprintf("tf-acc-test-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKubernetesDataSourceConfigMapConfig_basic(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.kubernetes_config_map.test", "metadata.0.name", name),
+					resource.TestCheckResourceAttrSet("data.kubernetes_config_map.test", "metadata.0.generation"),
+					resource.TestCheckResourceAttrSet("data.kubernetes_config_map.test", "metadata.0.resource_version"),
+					resource.TestCheckResourceAttrSet("data.kubernetes_config_map.test", "metadata.0.self_link"),
+					resource.TestCheckResourceAttrSet("data.kubernetes_config_map.test", "metadata.0.uid"),
+					resource.TestCheckResourceAttr("data.kubernetes_config_map.test", "metadata.0.annotations.%", "2"),
+					resource.TestCheckResourceAttr("data.kubernetes_config_map.test", "metadata.0.annotations.TestAnnotationOne", "one"),
+					resource.TestCheckResourceAttr("data.kubernetes_config_map.test", "metadata.0.annotations.TestAnnotationTwo", "two"),
+					resource.TestCheckResourceAttr("data.kubernetes_config_map.test", "metadata.0.labels.TestLabelOne", "one"),
+					resource.TestCheckResourceAttr("data.kubernetes_config_map.test", "metadata.0.labels.TestLabelTwo", "two"),
+					resource.TestCheckResourceAttr("data.kubernetes_config_map.test", "metadata.0.labels.TestLabelThree", "three"),
+					resource.TestCheckResourceAttr("data.kubernetes_config_map.test", "data.%", "2"),
+					resource.TestCheckResourceAttr("data.kubernetes_config_map.test", "data.one", "first"),
+					resource.TestCheckResourceAttr("data.kubernetes_config_map.test", "data.two", "second"),
+				),
+			},
+		},
+	})
+}
+
+func testAccKubernetesDataSourceConfigMapConfig_basic(name string) string {
+	return testAccKubernetesConfigMapConfig_basic(name) + `
```

Comment:
This tripped me up a little bit, as at first glance it seems like there's a bunch of checks in the test that aren't in the fixture.

I get that here we're trying to re-use the code from the `ResourceConfigMapConfig_basic` test. This makes sense to save writing code, but then we're creating a dependency between these two tests such that we have to keep their checks in sync when we update the underlying fixture. 

I find it's better to keep acceptance tests as explicit and isolated as possible, so that you can look at the test code and easily understand what is being tested and the expected behaviour without having to step through too much code. There is a pretty horrible acronym [DAMP](https://stackoverflow.com/a/11837973), as an antonym to DRY, that somewhat applies here.

What I would do here is explicitly code a `config_map` resource here, and maybe make it simpler so we don't have so many checks above. We are already testing all things that the `resourceKubernetesConfigMapRead` function does in that other test, so we don't need to re-test them here. 

edit: I just noticed you took this from the closed PR you linked, but I would have left the same comment on that PR 😄

Reply:
Thanks, I really like this idea! I honestly had a pretty hard time understanding what this code did and wasn't happy with it. But I simplified it now.